prog/encoding.go

```diff
@@ -564,11 +579,15 @@ func (p *parser) parseArgAddr(typ Type, dir Dir) (Arg, error) {
 		p.Parse('=')
 		if p.Char() == 'A' {
 			p.Parse('A')
-			p.Parse('N')
-			p.Parse('Y')
-			p.Parse('=')
-			anyPtr := p.target.getAnyPtrType(typ.Size())
-			typ, elem, elemDir = anyPtr, anyPtr.Elem, anyPtr.ElemDir
+			// If we have AUTO here for a struct, we don't want to parse it
+			// here but in parseArgImpl.
+			if p.Char() != 'U' {
```

Comment:
I think it'd be better to explicitly look ahead for more than 1 character instead. In the end, that's what we're doing here anyway.

Something like `if p.HasNext("ANY")`?

Reply:
Done in the first commit. It does make the main change (second commit) much cleaner :-)